app_flutter/lib/status_grid.dart

```diff
@@ -3,43 +3,61 @@
 // found in the LICENSE file.
 
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 
 import 'package:cocoon_service/protos.dart'
     show Commit, CommitStatus, Stage, Task;
 
+import 'state/flutter_build.dart';
 import 'commit_box.dart';
 import 'task_box.dart';
 
+/// Container that manages the layout and data handling for [StatusGrid].
+///
+/// If there's no data for [StatusGrid], it shows [CircularProgressIndicator].
+class StatusGridContainer extends StatelessWidget {
```

Comment:
Are there existing tests that cover this class? Would like to have a test that asserts that the progress indicator display when status is empty is correct

Reply:
Yes, they're in `status_grid_test.dart` and cover that case. Should they be moved to `status_grid_container_test.dart`?

Reply:
Nah, that is fine